conans/cli/api/subapi/install.py

```diff
@@ -59,3 +62,30 @@ def install_consumer(deps_graph, generators=None, no_imports=False, deploy=False
             deploy_conanfile.folders.set_base_imports(output_folder)
             if hasattr(deploy_conanfile, "deploy") and callable(deploy_conanfile.deploy):
                 run_deploy(deploy_conanfile, output_folder)
+
+    @staticmethod
+    def deploy_consumer(deps_graph, generators=None, source_folder=None, output_folder=None):
+        """ Once a dependency graph has been installed, there are things to be done, like invoking
+        generators for the root consumer, or calling imports()/deploy() to copy things to user space.
+        This is necessary for example for conanfile.txt/py, or for "conan install <ref> -g
+        """
+        root_node = deps_graph.root
+        conanfile = root_node.conanfile
+
+        conanfile.folders.set_base_source(source_folder)
+        conanfile.folders.set_base_imports(output_folder)
+        conanfile.folders.set_base_generators(output_folder)
+
+        if type(conanfile).system_requirements != ConanFile.system_requirements:
+            call_system_requirements(conanfile)
+
+        for r, d in conanfile.dependencies.items():
+            new_folder = os.path.join(output_folder, d.ref.name)
```

Comment:
To make it collision-safe and multi-config?
```suggestion
            new_folder = os.path.join(output_folder, d.ref.name, str(d.ref.version))
            if conanfile.settings.get_safe("arch"):
                new_folder = os.path.join(new_folder, str(conanfile.settings.arch))
            if conanfile.settings.get_safe("build_type"):
                new_folder = os.path.join(new_folder, str(conanfile.settings.build_type))
```

Reply:
This is the POC. The idea is that this will not be built-in here, but on the user side, something in the line of the ``layout()`` and we could provide something like ``cmake_layout()``, as ``typical_deploy()``. This is basically one of the points that we should discuss around this feature.